src/Facets/DeBridgeDlnFacet.sol

```diff
@@ -0,0 +1,182 @@
+        uint256 indexed destinationChainId,
+        bytes receiver
+    );
+
+    /// Constructor ///
+
+    /// @notice Constructor for the contract.
+    /// @param _dlnSource The address of the DLN order creation contract
+    constructor(IDlnSource _dlnSource) {
+        dlnSource = _dlnSource;
+    }
+
+    /// External Methods ///
+
+    /// @notice Bridges tokens via DeBridgeDLN
+    /// @param _bridgeData The core information needed for bridging
+    /// @param _deBridgeDlnData Data specific to DeBridgeDLN
+    function startBridgeTokensViaDeBridgeDln(
+        ILiFi.BridgeData memory _bridgeData,
+        DeBridgeDlnData calldata _deBridgeDlnData
+    )
+        external
+        payable
+        nonReentrant
+        refundExcessNative(payable(msg.sender))
+        validateBridgeData(_bridgeData)
+        doesNotContainSourceSwaps(_bridgeData)
+        doesNotContainDestinationCalls(_bridgeData)
+    {
+        LibAsset.depositAsset(
+            _bridgeData.sendingAssetId,
+            _bridgeData.minAmount
+        );
+        _startBridge(_bridgeData, _deBridgeDlnData);
+    }
+
+    /// @notice Performs a swap before bridging via DeBridgeDLN
+    /// @param _bridgeData The core information needed for bridging
+    /// @param _swapData An array of swap related data for performing swaps before bridging
+    /// @param _deBridgeDlnData Data specific to DeBridgeDLN
+    function swapAndStartBridgeTokensViaDeBridgeDln(
+        ILiFi.BridgeData memory _bridgeData,
+        LibSwap.SwapData[] calldata _swapData,
+        DeBridgeDlnData calldata _deBridgeDlnData
+    )
+        external
+        payable
+        nonReentrant
+        refundExcessNative(payable(msg.sender))
+        containsSourceSwaps(_bridgeData)
+        doesNotContainDestinationCalls(_bridgeData)
+        validateBridgeData(_bridgeData)
+    {
+        uint256 fee = dlnSource.globalFixedNativeFee();
+        address assetId = _bridgeData.sendingAssetId;
+        _bridgeData.minAmount = _depositAndSwap(
+            _bridgeData.transactionId,
+            _bridgeData.minAmount,
+            _swapData,
+            payable(msg.sender),
+            LibAsset.isNativeAsset(assetId) ? 0 : fee
+        );
+        _startBridge(_bridgeData, _deBridgeDlnData);
+    }
+
+    /// Internal Methods ///
+
+    /// @dev Contains the business logic for the bridge via DeBridgeDLN
+    /// @param _bridgeData The core information needed for bridging
+    /// @param _deBridgeDlnData Data specific to DeBridgeDLN
+    function _startBridge(
+        ILiFi.BridgeData memory _bridgeData,
+        DeBridgeDlnData calldata _deBridgeDlnData
+    ) internal {
+        uint256 fee = dlnSource.globalFixedNativeFee();
+        IDlnSource.OrderCreation memory orderCreation = IDlnSource
+            .OrderCreation({
+                giveTokenAddress: _bridgeData.sendingAssetId,
+                giveAmount: _bridgeData.minAmount,
+                takeTokenAddress: abi.encodePacked(
```

Comment:
This encoding could be done in the backend and instead of address parameter we use bytes parameter. Saves the (unnecessary) encoding step on-chain